ops/internal/config/superchain.go

```diff
@@ -27,6 +41,64 @@ func MustParseSuperchain(in string) Superchain {
 	return sup
 }
 
+// FindValidL1URL finds a valid l1-rpc-url for a given superchain by finding matching l1 chainId
+func FindValidL1URL(lgr log.Logger, urls []string, superchain Superchain) (string, error) {
+	for i, url := range urls {
+		url = strings.TrimSpace(url)
+		if url == "" {
+			continue
+		}
+
+		if err := validateL1ChainID(url, superchain); err != nil {
+			lgr.Warn("l1-rpc-url has mismatched l1 chainId", "urlIndex", i)
+			continue
+		}
+
+		lgr.Info("l1-rpc-url has matching l1 chainId", "urlIndex", i)
+		return url, nil
+	}
+	return "", fmt.Errorf("no valid L1 RPC URL found for superchain %s", superchain)
+}
+
+// validateL1ChainID checks if the l1RpcUrl has the expected chain ID for the superchain
+func validateL1ChainID(l1RpcUrl string, superchain Superchain) error {
+	expectedChainID, ok := SuperchainChainIds[superchain]
+	if !ok {
+		return fmt.Errorf("unknown superchain: %s", superchain)
+	}
+
+	chainID, err := getL1ChainId(l1RpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to get chainId from l1RpcUrl: %w", err)
+	}
+
+	if chainID != expectedChainID {
+		return fmt.Errorf("l1RpcUrl chainId mismatch: got %d, expected %d for superchain %s",
+			chainID, expectedChainID, superchain)
+	}
+
+	return nil
+}
+
+// getL1ChainId connects to an Ethereum RPC endpoint and retrieves its chain ID
+func getL1ChainId(rpcURL string) (uint64, error) {
+	client, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		return 0, fmt.Errorf("failed to connect to L1 RPC: %w", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Don't declare context like this in private methods, instead have the caller pass them in